### PR TITLE
Adding Namespace level configuration for all litmus-portal components.

### PR DESCRIPTION
**Pull request by ishangupta-ds**
## Proposed changes

This PR removes hardcoded namespaces and also updates manifests for using configured namespace.

## Types of changes

What types of changes does your code introduce to Litmus? Put an `x` in the boxes that apply
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices applies)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
- [x] I have read the [CONTRIBUTING](https://github.com/litmuschaos/litmus/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the commit for DCO to be passed.
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have added necessary documentation (if appropriate)

## Dependency
- None.

## Special notes for your reviewer:
This update is needed urgently.

### CODE REVIEW

File: litmus-portal/graphql-server/manifests/subscriber.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v1
```
Comment (gdsoumya) [inline litmus-portal/graphql-server/manifests/subscriber.yml:1]: remove the unnecessary changes, the yaml was in a minified form but your editor has formatted. Just keep the namespace changes and remove the other formatting changes.

File: litmus-portal/graphql-server/pkg/k8s/custer.go [hunk 2]
```diff
@@ -58,6 +58,14 @@ func CreateDeployment(namespace, token string) (*appsv1.Deployment, error) {
 									Name:  "TOKEN",
 									Value: token,
 								},
+								{
+									Name:  "NameSpace",
```
Comment (gdsoumya) [inline litmus-portal/graphql-server/pkg/k8s/custer.go]: Change to uppercase for env var name

File: litmus-portal/k8s-manifest.yml [hunk 3]
```diff
@@ -161,7 +154,7 @@ spec:
     - ReadWriteOnce
   resources:
     requests:
-      storage: 20Gi
+      storage: 4Gi
```
Comment (gdsoumya) [inline litmus-portal/k8s-manifest.yml:157]: Why was this changed? It is too low for production

File: litmus-portal/k8s-manifest.yml [hunk 4]
```diff
@@ -1,9 +1,3 @@
----
-apiVersion: v1
```
Comment (gdsoumya) [inline litmus-portal/k8s-manifest.yml:2]: Why is this removed? The litmus ns needs to be created separately then.

File: litmus-portal/graphql-server/pkg/k8s/custer.go [hunk 5]
```diff
@@ -8,7 +8,7 @@ import (
 	"k8s.io/client-go/kubernetes"
 
 	"k8s.io/client-go/tools/clientcmd"
-
+	corev1 "k8s.io/api/core/v1"
```
Comment (gdsoumya) [inline litmus-portal/graphql-server/pkg/k8s/custer.go]: Duplicate import. `apiv1` already refers to the same import

**Review by gdsoumya — CHANGES_REQUESTED** [review]

File: litmus-portal/graphql-server/manifests/subscriber.yml (hunk 1 above)
Comment (ishangupta-ds) [reply litmus-portal/graphql-server/manifests/subscriber.yml:1]: okay sure.

File: litmus-portal/k8s-manifest.yml (hunk 3 above)
Comment (ishangupta-ds) [reply litmus-portal/k8s-manifest.yml:157]: Yeah removing it, just for okteto this was needed.

File: litmus-portal/k8s-manifest.yml (hunk 4 above)
Comment (ishangupta-ds) [reply litmus-portal/k8s-manifest.yml:2]: I have added it back, we will have a custom script for okteto. to make changes in our portal yaml.

**Comment by ishangupta-ds** [comment]
@gdsoumya PTAL made the suggested changes.

File: litmus-portal/graphql-server/manifests/cluster-subscriber.yml [hunk 6]
```diff
@@ -405,6 +405,10 @@ spec:
             - server
           image: argoproj/argocli:v2.9.3
           name: argo-server
+          resources:
```
Comment (gdsoumya) [inline litmus-portal/graphql-server/manifests/cluster-subscriber.yml:408]: remove limits

**Review by gdsoumya — CHANGES_REQUESTED** [review]
There will be no workflow NS it will be either specified in the workfow manifest, or will be defaulted to subscriber NS

File: litmus-portal/graphql-server/manifests/cluster-subscriber.yml (hunk 6 above)
Comment (ishangupta-ds) [reply litmus-portal/graphql-server/manifests/cluster-subscriber.yml:408]: Actually these limits have been requested by @rajdas98 @ksatchit and @ajeshbaby ... we can increase the value here but its needed i guess.. @gdsoumya

File: litmus-portal/cluster-agents/subscriber/pkg/cluster/events/workflow.go [hunk 7]
```diff
@@ -18,6 +18,11 @@ const (
 	resyncPeriod time.Duration = 0
 )
 
+var (
+	DefaultScope = os.Getenv("WORKFLOW_SCOPE")
```
Comment (gdsoumya) [inline litmus-portal/cluster-agents/subscriber/pkg/cluster/events/workflow.go:22]: change to agent_scope and agent_ns

File: litmus-portal/cluster-agents/subscriber/pkg/gql/subscriptions.go [hunk 8]
```diff
@@ -87,7 +87,7 @@ func ClusterConnect(clusterData map[string]string) {
 			SendPodLogs(clusterData, podRequest)
 		} else if strings.Index("create update delete get", strings.ToLower(r.Payload.Data.ClusterConnect.Action.RequestType)) >= 0 {
 			logrus.Print("WORKFLOW REQUEST ", r.Payload.Data.ClusterConnect.Action)
-			_, err = k8s.ClusterOperations(r.Payload.Data.ClusterConnect.Action.K8SManifest, r.Payload.Data.ClusterConnect.Action.RequestType)
+			_, err = k8s.ClusterOperations(r.Payload.Data.ClusterConnect.Action.K8SManifest, r.Payload.Data.ClusterConnect.Action.RequestType, r.Payload.Data.ClusterConnect.Action.Namespace)
```
Comment (gdsoumya) [inline litmus-portal/cluster-agents/subscriber/pkg/gql/subscriptions.go:90]: No extra parameter is required, either the user specifies the ns in the workflow manifest or we default to agent ns

File: litmus-portal/cluster-agents/subscriber/pkg/k8s/operations.go [hunk 9]
```diff
@@ -138,7 +140,7 @@ func applyRequest(requestType string, obj *unstructured.Unstructured) (*unstruct
 }
 
 // This function handles cluster operations
-func ClusterOperations(manifest string, requestType string) (*unstructured.Unstructured, error) {
+func ClusterOperations(manifest string, requestType string, namespace string) (*unstructured.Unstructured, error) {
```
Comment (gdsoumya) [inline litmus-portal/cluster-agents/subscriber/pkg/k8s/operations.go:143]: No extra parameter required

File: litmus-portal/graphql-server/manifests/cluster-subscriber.yml (hunk 6 above)
Comment (gdsoumya) [reply litmus-portal/graphql-server/manifests/cluster-subscriber.yml:408]: We haven't properly evaluated the resource requirements, these values might not work in production with multiple workflows and resources. And moreover a user might have different resource constraints, he can change it as required.

File: litmus-portal/graphql-server/pkg/file_handlers/file_handlers.go [hunk 10]
```diff
@@ -17,6 +17,9 @@ func FileHandler(w http.ResponseWriter, r *http.Request) {
 	serviceAddr := os.Getenv("SERVICE_ADDRESS")
 	subscriberImage := os.Getenv("SUBSCRIBER_IMAGE")
 	subscriberNS := os.Getenv("DEPLOYER_NAMESPACE")
+	subscriberSC := os.Getenv("SUBSCRIBER_SCOPE")
+	workflowSC := os.Getenv("WORKFLOW_SCOPE")
```
Comment (gdsoumya) [inline litmus-portal/graphql-server/pkg/file_handlers/file_handlers.go:21]: Subscriber and workflow will have the same scope and ns. If there is any change for the workflow it will be mentioned in the workflow yaml itself.

**Review by gdsoumya — CHANGES_REQUESTED** [review]

File: litmus-portal/cluster-agents/subscriber/pkg/cluster/events/workflow.go (hunk 7 above)
Comment (ishangupta-ds) [reply litmus-portal/cluster-agents/subscriber/pkg/cluster/events/workflow.go:22]: sure..

**Comment by rajdas98** [comment]
changes:
- Remove cluster scope YAML and add only namespace scope YAML(no need to change the name for namespace scope yaml)
- Add good resource limit to the self deployer.
